content/blog/2021/07/2021-07-29-introducing-conventional-commits-plugin-for-jenkins.adoc

```diff
@@ -0,0 +1,168 @@
+---
+:layout: post
+:title: "Introducing the Conventional Commits Plugin for Jenins"
+:tags:
+- gsoc
+- gsoc2021
+- conventionalcommits
+- plugin
+:author: adi10hero
+:sig: gsoc
+opengraph:
+  image: /images/gsoc/2021/conventional-commits/conventionalCommitsPluginForJenkins.png
+links:
+  discourse: https://community.jenkins.io/t/introducing-the-conventional-commits-plugin/272
+---
+
+image:/images/gsoc/2021/conventional-commits/conventionalCommitsPluginForJenkins.png[GSoC, height=420, role=center, float=center]
+
+The conventional commits plugin is a Google Summer of Code project.
+Special thanks to the mentors Gareth Evans, Kristin Whetstone, Olivier Vernin and Allan Burdajewicz.
+
+=== What are Conventional Commits
+
+According to the link:https://www.conventionalcommits.org/[official website], conventonal commits are, "A specification for adding human and machine readable meaning to commit messages."
+
+Conventional commits are a lightweight convention on top of commit messages.
+
+The following table shows major structural elements offered by the conventional commits convention.
+
+|===
+|Structural Element|Example
+
+|Chore
+|`chore: improve logging`
+
+|Fix
+|`fix: minor bug fix`
+
+|Feat
+|`feat: add a new feature`
+
+|Breaking Change
+|`BREAKING CHANGE: reimplement`
+|===
+
+=== Why Conventional Commits
+
+As the CI/CD world is moving more towards complete automation and minimal human interaction, the ability to fully automate a release is desired.
+Conventional Commits enable the use of automated systems on top of commit messages.
+These systems can "truly" automate a release with almost no human interaction. 
+
+The convention dovetails with link:https://semver.org/[semantic versioning].
+Let's take an example, a maven project is currently versioned at `1.2.0`.
+The following table shows how conventional commits would bump the version depending on the type of the commit.
+
+|===
+|Commit Message|Version Bump|SemVer Equivalent
+
+|`chore: improve logging`
+|`1.2.0` -> `1.2.0`
```

Comment:
I would have expected this to bump to `1.2.1` - although, with JEP-229 a release would not be created automatically.